chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/impl/FilterManagerImpl.java

```diff
@@ -0,0 +1,125 @@
+
+import gov.healthit.chpl.dao.FilterDAO;
+import gov.healthit.chpl.dto.FilterDTO;
+import gov.healthit.chpl.dto.FilterTypeDTO;
+import gov.healthit.chpl.exception.EntityRetrievalException;
+import gov.healthit.chpl.exception.ValidationException;
+import gov.healthit.chpl.manager.FilterManager;
+
+@Component("filterManager")
+public class FilterManagerImpl extends SecuredManager implements FilterManager {
+    private FilterDAO filterDAO;
+
+    @Autowired
+    public FilterManagerImpl(final FilterDAO filterDAO) {
+        this.filterDAO = filterDAO;
+    }
+
+    @Override
+    @Transactional
+    @PostFilter("@permissions.hasAccess(T(gov.healthit.chpl.permissions.Permissions).FILTER, "
+            + "T(gov.healthit.chpl.permissions.domains.FilterDomainPermissions).GET_BY_FILTER_TYPE, filterObject)")
+    public List<FilterDTO> getByFilterType(FilterTypeDTO filterTypeDTO) {
+        return filterDAO.getByFilterType(filterTypeDTO);
+    }
+
+    @Override
+    @Transactional
+    @PreAuthorize("@permissions.hasAccess(T(gov.healthit.chpl.permissions.Permissions).FILTER, "
+            + "T(gov.healthit.chpl.permissions.domains.FilterDomainPermissions).CREATE, #filterDTO)")
+    public FilterDTO create(FilterDTO filterDTO) throws EntityRetrievalException, ValidationException {
+        Set<String> errors = validateForCreate(filterDTO);
+        if (errors.size() > 0) {
+            throw new ValidationException(errors, null);
+        }
+        return filterDAO.create(filterDTO);
+    }
+
+    @Override
+    @Transactional
+    @PreAuthorize("@permissions.hasAccess(T(gov.healthit.chpl.permissions.Permissions).FILTER, "
+            + "T(gov.healthit.chpl.permissions.domains.FilterDomainPermissions).UPDATE, #filterDTO)")
+    public FilterDTO update(FilterDTO filterDTO) throws EntityRetrievalException, ValidationException {
+        Set<String> errors = validateForUpdate(filterDTO);
+        if (errors.size() > 0) {
+            throw new ValidationException(errors, null);
+        }
+        return filterDAO.update(filterDTO);
+    }
+
+    @Override
+    @Transactional
+    @PreAuthorize("@permissions.hasAccess(T(gov.healthit.chpl.permissions.Permissions).FILTER, "
+            + "T(gov.healthit.chpl.permissions.domains.FilterDomainPermissions).DELETE, #filterDTO)")
+    public void delete(FilterDTO filterDTO) throws EntityRetrievalException, ValidationException {
+        Set<String> errors = validateForDelete(filterDTO);
+        if (errors.size() > 0) {
+            throw new ValidationException(errors, null);
+        }
+        filterDAO.delete(filterDTO);
+    }
+
+    @Override
+    @Transactional
+    public FilterTypeDTO getFilterType(Long filterTypeId) throws EntityRetrievalException {
+        return filterDAO.getFilterTypeById(filterTypeId);
+    }
+
+    @Override
+    @Transactional
+    public FilterDTO getByFilterId(Long filterId) throws EntityRetrievalException {
+        return filterDAO.getById(filterId);
```

Comment:
Since there is no security on this, could anyone calling this code see any other user's filter? I know we don't have the concept of public and private but I was thinking they were all sort of private. Should we have permissions that only let a user get to a filter if they are the user_id on it? If this was intentional then that's fine.